pyro/nn/module.py

```diff
@@ -579,7 +635,11 @@ def __getattr__(self, name):
 
         return result
 
-    def __setattr__(self, name, value):
+    def __setattr__(
+        self,
+        name: str,
+        value: Union[torch.Tensor, torch.nn.Module, PyroParam, PyroSample],
```

Comment:
`value` should have type `Any`, since we fall back to `super.__setattr__(name, value)` at the last line of this method.

Reply:
Interesting. Is that why mypy thinks `torch.nn.Module`s attributes are either `Tensor` or `Module` because they annotated them as such 🤔 ? That always confused me.